packages/contracts/contracts/foundry/SphinxUtils.sol

```diff
@@ -929,6 +934,114 @@ contract SphinxUtils is SphinxConstants, StdUtils {
         return count;
     }
 
+    function serializeFoundryDeploymentInfo(
+        FoundryDeploymentInfo memory _deployment
+    ) public returns (string memory) {
+        // Set the object key to an empty JSON, which ensures that there aren't any existing values
+        // stored in memory for the object key.
+        vm.serializeJson(deploymentInfoKey, "{}");
+
+        // Serialize simple fields
+        vm.serializeAddress(deploymentInfoKey, "safeAddress", _deployment.safeAddress);
+        vm.serializeAddress(deploymentInfoKey, "moduleAddress", _deployment.moduleAddress);
+        vm.serializeAddress(deploymentInfoKey, "executorAddress", _deployment.executorAddress);
+        vm.serializeBytes(deploymentInfoKey, "safeInitData", _deployment.safeInitData);
+        vm.serializeBool(deploymentInfoKey, "requireSuccess", _deployment.requireSuccess);
+        vm.serializeString(
+            deploymentInfoKey,
+            "sphinxLibraryVersion",
+            _deployment.sphinxLibraryVersion
+        );
+        vm.serializeBool(deploymentInfoKey, "arbitraryChain", _deployment.arbitraryChain);
+        vm.serializeBytes(
+            deploymentInfoKey,
+            "encodedAccountAccesses",
+            _deployment.encodedAccountAccesses
+        );
+        // Next, we'll serialize `uint` values as ABI encoded bytes. We don't serialize them as
+        // numbers to prevent the possibility that they lose precision due JavaScript's relatively
+        // low integer size limit. We'll ABI decode these values in TypeScript. It'd be simpler to
+        // serialize the numbers as strings without ABI encoding them, but that strategy is blocked
+        // by a Foundry bug: https://github.com/foundry-rs/foundry/issues/6533
+        vm.serializeBytes(deploymentInfoKey, "nonce", abi.encode(_deployment.nonce));
```

Comment:
It’s probably worth thinking about if there’s a way we can prevent ourselves from introducing a new `uint` in the future which is not handled this way. I can’t think of anything off the top of my head, but I’m sure there’s something we could do. I don’t think it’s a huge deal, but it is definitely a foot-gun.

Reply:
Agreed it's a footgun. I think it'd be pretty straightforward to check that there aren't any numbers in the `DeploymentInfo` object in TypeScript, but IMO it's not a big deal, since I think a mistake like this is unlikely to make it past both of us. Feel free to make a ticket if you disagree